metapool/sample_sheet.py

```diff
@@ -1237,38 +1237,99 @@ def __init__(self, path=None):
         }
 
 
+def _parse_header(fp):
+    df = pd.read_csv(fp, dtype="str", sep=",", header=None,
+                     names=range(100))
+
+    # pandas will have trouble reading a sample-sheet if the csv has a
+    # variable number of columns. This occurs in legacy sheets when a user
+    # has introduced one too many ',' characters in a line.
+    #
+    # the solution is to fix the number of initial columns at a high enough
+    # value to include all columns, name them with integers, and later
+    # truncate all columns that are entirely empty.
+    df.dropna(how='all', axis=1, inplace=True)
+
+    # remove all whitespace rows (drop all rows that are entirely empty)
+    df.dropna(how='all', axis=0, inplace=True)
+
+    # remove all comments rows, whether they are at the top of the file (no
+    # longer supported, technically), or not.
+    comment_rows = df.index[df[0].str.startswith("#")].tolist()
+    df = df.drop(index=comment_rows)
+    # reset the index to make it easier to post-process.
+    df.reset_index(inplace=True, drop=True)
+
+    # for simplicity's sake, assume the first row marks the [Header]
+    # column and raise an Error if not. By convention it should be, once
+    # legacy comments and whitespace rows are removed.
+    if df[0][0] != '[Header]':
+        raise ValueError("Top section is not [Header]")
```

Comment:
This is not true for legacy files, right? I think I have seen some with comments.

Reply:
Yup, right before Yoshiki left he started that push to drop support for those. I believe it's because the info was usually metadata that now needed to go in the Header and Contact sections. What the current code does is ignore lines beginning with '#' and that's what we do here as well. Once we find and drop all comment rows (L1258-L1259) then ostensibly [Header] will be the first line of the sample-sheet.

Reply:
Got it, just to confirm: this means that old sample-sheets with comments will still be supported; right?